examples/plotting/file/line_toggle.py

```diff
@@ -0,0 +1,39 @@
+"""
+Example demonstrating the picking of line objects. The transparency of
+the lines is toggled when clicked.
+"""
+
+import numpy as np
+from bokeh.plotting import output_file, show, figure, Figure
+from bokeh.models import TapTool, CustomJS
+
+# The use of the tap tool causes Bokehjs to "select" a line when it is
+# clicked on, which is not what we want in this example; we want to
+# toggle a visual property of the line ourselves. To work arount this,
+# we modify the selection_glyph and nonselection_glyph as well.
```

Comment:
I think we will have to crystalize a much better/tighter "event" notion. I am not sure what this should look like. Perhaps you and @brendancol can collaborate on ideas? Additionally @havocp might have thoughts. Some things that are currently "model state changes" should really be "events" (e.g., a button was pushed).